tests/bootstrap_project/test/project_test.dart

```diff
@@ -0,0 +1,432 @@
+        });
+
+        test('has a pubspec file', () {
+          expect(
+            File(path.join(flutterDir, 'pubspec.yaml')).existsSync(),
+            isTrue,
+            reason: 'Flutter pubspec file does not exist.',
+          );
+        });
+
+        test('has a main file', () {
+          expect(
+            File(path.join(flutterDir, 'lib', 'main.dart')).existsSync(),
+            isTrue,
+            reason: 'Flutter main file does not exist.',
+          );
+        });
+      });
+
+      group('then the client project', () {
+        test('folder is created', () {
+          expect(
+            Directory(clientDir).existsSync(),
+            isTrue,
+            reason: 'Client folder does not exist.',
+          );
+        });
+
+        test('has a pubspec file', () {
+          expect(
+            File(path.join(clientDir, 'pubspec.yaml')).existsSync(),
+            isTrue,
+            reason: 'Client pubspec file does not exist.',
+          );
+        });
+
+        test('has a project_client file', () {
+          expect(
+            File(
+              path.join(clientDir, 'lib', '${projectName}_client.dart'),
+            ).existsSync(),
+            isTrue,
+            reason: 'Client project_client file does not exist.',
+          );
+        });
+
+        test('has a protocol client file', () {
+          expect(
+            File(
+              path.join(clientDir, 'lib', 'src', 'protocol', 'client.dart'),
+            ).existsSync(),
+            isTrue,
+            reason: 'Client protocol client file does not exist.',
+          );
+        });
+      });
+    });
+  }, timeout: timeout);
+
+  group('Given a clean state', () {
+    final (projectName, commandRoot) = createRandomProjectName(tempPath);
+    final (serverDir, _, clientDir) = createProjectFolderPaths(projectName);
+
+    late Process createProcess;
+
+    tearDown(() async {
+      print('----------------------------------------------------------');
+      createProcess.kill();
+      await Process.run(
+        'docker',
+        ['compose', 'down', '-v'],
+        workingDirectory: commandRoot,
+      );
+    });
+
+    test(
+        'when removing generated files from a new project and running generate then the files are recreated successfully',
+        () async {
+      createProcess = await Process.start(
+        'serverpod',
+        ['create', projectName, '-v'],
+        workingDirectory: tempPath,
+        environment: {
+          'SERVERPOD_HOME': rootPath,
+        },
+      );
+
+      createProcess.stdout.transform(Utf8Decoder()).listen(print);
+      createProcess.stderr.transform(Utf8Decoder()).listen(print);
+
+      var createProjectExitCode = await createProcess.exitCode;
+      expect(createProjectExitCode, 0);
+
+      // Delete generated files
+      await Process.run('rm', ['-f', '${serverDir}/lib/src/generated/*.dart']);
+      await Process.run('rm', ['-f', '${clientDir}/lib/src/protocol/*.dart']);
```

Comment:
Note - We should add the yaml endpoint file we accidentally removed to this test in that PR.

Reply:
indeed but need to have the change on main!

Reply:
You merged it into main now right? So now you can add it to this test :)

PR that was merged: https://github.com/serverpod/serverpod/commit/885b2c2a625b223ba61f51d65434df9e917d15fd